### DISCUSSION TITLE
Type for query with 0 parameters

### DISCUSSION THREAD

**Discussion by ChuckJonas** [reduxjs/redux-toolkit#1548] [Q&A] [answered]
I'm trying to figure out how to define a query that has no parameters.

I can do something like this:

```ts
getPosts: builder.query<WorkItem[], void>({ ... })
```

and I can then call `useGetPostsQuery()`.

However, I cannot pass the second parameter for options if I do this:

```ts
useGetPostsQuery({refetchOnFocus: true})
//Argument of type '{ refetchOnFocus: boolean; }' is not assignable to parameter of type 'void | unique symbol'.
  Type '{ refetchOnFocus: boolean; }' is not assignable to type 'unique symbol
```

Is there a proper way to define a query with any params?

**Comment by markerikson** [comment, marked as answer]
You _always_ have to pass a "query cache key" as the first arg to a `useQuery()` hook.  In this case, the correct cache key would be `undefined`.  So: `useGetPostsQuery(undefined, {refetchOnFocus: true})`.
